src/poetry/console/commands/self/update.py

```diff
@@ -88,22 +88,114 @@ def pool(self) -> Pool:
         return pool
 
     def handle(self) -> int:
-        from poetry.core.packages.dependency import Dependency
+        from poetry.utils._compat import metadata
+
+        poetry_release = self._find_poetry_release()
+        core_release = self._find_compatible_core_release(poetry_release)
+
+        update_releases = [
+            release
+            for release in {poetry_release, core_release}
+            if release.version.text != metadata.version(release.name)
+        ]
+
+        if not update_releases:
+            self.line("You are using the latest version")
+            return 0
+
+        msg = [
+            f"<c1>{release.pretty_name}</c1> to <c2>{release.version}</c2>"
+            for release in update_releases
+        ]
+        self.line(f"Updating {' and '.join(msg)}.")
+        self.line("")
+
+        self.update(update_releases)
+
+        if poetry_release in update_releases:
+            self._make_bin()
+
+        self.line("")
+        self.line(f"Successfully updated {' and '.join(msg)}. Great!")
+
+        return 0
+
+    def update(self, releases: list[Package]) -> None:
+        from poetry.utils.env import EnvManager
+
+        env = EnvManager.get_system_env(naive=True)
+
+        # We can't use is_relative_to() since it's only available in Python 3.9+
```

Comment:
This comment could be a little more descriptive -- we're throwing away the result of the `env.path` call because it throws an exception when the paths are not adjacent, right?